### PR TITLE
Switch create-redwood-app template to TypeScript

### PR DESCRIPTION
**Pull request by peterp**
We would like to give people the option to generate a TypeScript project. This PR will allow people to create both a JS and a TS RedwoodJS project. JavaScript will always be the default.

- [x] Create a working TypeScript template.
- [x] Convert the TypeScript project to a JavaScript project.
- [x] Use prettier to clean up the output.
- [x] Add a flag to not convert the TypeScript project to TypeScript.

### CODE REVIEW

**Comment by jvanbaarsen** [comment]
@peterp Does that mean that you "need" to use typescript when using Redwood? Or will it remain optional?

**Comment by peterp** [comment]
@jvanbaarsen 100% optional. It'll always be JavaScript by default.

File: packages/internal/src/ts2js.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+/**
+ * Get all the source code from a Redwood application.
+ */
+export const getSourceFiles = (cwd = getPaths().base) => {
+  return glob.sync('**/src/**/**.{ts,js,tsx,jsx}', {
+    cwd,
+  })
+}
+
+/**
```
Comment (papaponmx) [inline packages/internal/src/ts2js.ts:19]: Brilliant 👏👏

**Comment by papaponmx** [comment]
It looks like the only thing missing is to solve the conflicts, right?

**Comment by thedavidprice** [comment]
adding @dac09 as a reviewer to confirm changes he made to CRWA template are reflected in the final changes here

**Comment by dac09** [comment]
Just needs:
- App.js -> App.tsx after the changes in the last release.
- Do we worry about custom index being JS in the TS version of the project when running `yarn rw setup custom-web-index`?

**Comment by peterp** [comment]
@dac09 I'm sorry, I rewrote the `test-tutorial` function. It was failing for me when I was trying to make this work. I wanted to have it a bit more composable, so I broke each task into a separate function.

You can now do something like this:

```
./tasks/run-cypress /tmp/path --no-copy --no-start
```

https://github.com/redwoodjs/redwood/pull/1810/files#diff-e4d8d48827bb7da131ab93737cf7a718801e74e8c86b22e011f27dd4f948b1edR134-R143

/cc @thedavidprice

File: tasks/run-e2e [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      'yarn babel-node src/create-redwood-app.js',
+      [REDWOOD_PROJECT_DIRECTORY, '--no-yarn-install', '--no-javascript'],
+      {
+        cwd: path.join(REDWOODJS_FRAMEWORK_PATH, 'packages/create-redwood-app'),
+        shell: true,
+        stdio: 'inherit',
+      }
+    )
+  } catch (e) {
+    console.error('Could not build RedwoodJS framework')
```
Comment (peterp) [inline tasks/run-e2e:51]:
```suggestion
    console.error('Could not create RedwoodJS project')
```

File: packages/cli/src/commands/generate/scaffold/scaffold.js [hunk 3]
```diff
@@ -382,7 +382,7 @@ export const routes = async ({ model: name, path: scaffoldPath = '' }) => {
 }
 
 const addScaffoldImport = () => {
-  const appJsPath = path.join(getPaths().web.src, 'App.js')
+  const appJsPath = path.join(getPaths().web.app)
```
Comment (peterp) [inline packages/cli/src/commands/generate/scaffold/scaffold.js:385]:
```suggestion
  const appJsPath = getPaths().web.app
```

File: packages/internal/src/index.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export * from './paths'
 export * from './config'
 export * from './dev'
+
```
Comment (peterp) [inline packages/internal/src/index.ts:4]:
```suggestion
```

**Pull request merged**